crates/build/src/lib.rs

```diff
@@ -12,24 +12,42 @@ use subprocess::{Exec, Redirection};
 use crate::manifest::{BuildAppInfoAnyVersion, RawComponentManifest};
 
 /// If present, run the build command of each component.
-pub async fn build(manifest_file: &Path) -> Result<()> {
+pub async fn build(manifest_file: &Path, component_id: &Option<String>) -> Result<()> {
     let manifest_text = tokio::fs::read_to_string(manifest_file)
         .await
         .with_context(|| format!("Cannot read manifest file from {}", manifest_file.display()))?;
     let app = toml::from_str(&manifest_text).map(BuildAppInfoAnyVersion::into_v1)?;
     let app_dir = parent_dir(manifest_file)?;
 
-    if app.components.iter().all(|c| c.build.is_none()) {
-        println!("No build command found!");
-        return Ok(());
-    }
+    match component_id {
+        None => {
+            if app.components.iter().all(|c| c.build.is_none()) {
+                println!("No build command found!");
+                return Ok(());
+            }
 
-    app.components
-        .into_iter()
-        .map(|c| build_component(c, &app_dir))
-        .collect::<Result<Vec<_>, _>>()?;
+            app.components
+                .into_iter()
+                .map(|c| build_component(c, &app_dir))
+                .collect::<Result<Vec<_>, _>>()?;
+
+            println!("Successfully ran the build command for the Spin components.");
+        }
+        Some(id) => {
+            if let Some(component) = app.components.into_iter().find(|c| &c.id == id) {
+                if component.build.is_some() {
+                    build_component(component, &app_dir)?;
+                    println!("Successfully ran the build command for component '{id}'.");
+                } else {
+                    eprintln!("Component '{id}' does not have a build command.");
+                    eprintln!("For information on specifying a build command, see https://developer.fermyon.com/spin/build#setting-up-for-spin-build.");
```

Comment:
Should this return an error result? That would signal to doctor that the fix didn't work, and might be helpful for other kinds of automation.

Reply:
The doctor should never call this for components without a build command - it checks before recommending the "build" treatment.

Yeah I dithered about the error result.  I think I kind of reached the conclusion that it hadn't _failed_, there was just _nothing to do._  And so I was a bit wary about Rust printing `Error: ...` which we can't currently stop it doing.  But if other folks would prefer an error than I'm happy to go along with that.

Reply:
Looking at the existing flow, "no components have a build command" is currently not an error (and prints to stdout rather than stderr so I am being inconsistent there at least).

Reply:
I am reworking it as you suggested to take a list of components so this will unify it to the existing flow.  I will use that existing flow for now; we can change it to be an error if we want.